tooling/metamodel-generator/src/main/java/org/hibernate/jpamodelgen/annotation/AnnotationMetaEntity.java

```diff
@@ -1014,11 +1110,17 @@ private void validateUpdateHql(
 			@Nullable TypeMirror returnType,
 			AnnotationMirror mirror,
 			AnnotationValue value) {
+		boolean reactive = usingReactiveSession(sessionType);
 		if ( returnType == null
-				|| returnType.getKind() != TypeKind.VOID
-				&& returnType.getKind() != TypeKind.INT ) {
+				|| (!reactive 
+						&& returnType.getKind() != TypeKind.VOID
+						&& returnType.getKind() != TypeKind.INT)
+				|| (reactive 
+						// for reactive calls, don't use the returnType param, which has been ununi-ed, we want to check the full one
+						&& !method.getReturnType().toString().equals(Constants.UNI_VOID)
+						&& !method.getReturnType().toString().equals(Constants.UNI_INTEGER)) ) {
```

Comment:
probably makes sense to extract a method for this long boolean condition.

Reply:
And also I'm quite ashamed of the type test 🤦